docs/madr/decisions/057-rework-mesh-routes-matching.md

```diff
@@ -0,0 +1,741 @@
+# Rework of Mesh*Routes matching
+
+* Status: accepted
+
+Technical story: https://github.com/kumahq/kuma/issues/10247
+
+## Context and Problem Statement
+
+At the moment when targeting `Mesh*Route` you need to put it in topLevel targetRef. 
+We want to iterate over policy design and move `Mesh*Route` to `spec.to[].targetRef` section follow model where `spec.targetRef`
+selects whole proxies, `spec.from[].targetRef` selects inbounds and `spec.to[].targetRef` selects outbounds
+
+## Decision Outcome
+
+- add possibility to put `Mesh*Routes` in `spec.to[].targetRef`
+
+## Considered options
+
+1. add possibility to put `Mesh*Routes` in `spec.to[].targetRef` -> `spec.targerRef` selects proxies, `spec.from[].targetRef` selects inbounds, `spec.to[].targetRef` selects outbounds.
+2. leave `Mesh*Routes` in top level targetRef as it is now, and introduce `spec.workloadTargetRef` to match whole proxies
+
+### Add possibility to put Mesh*Routes in `spec.to[].targetRef`
+
+#### Advantages
+
+- simplifying policy api as `spec.targetRef` will always pick proxy and `spec.to[].targetRef` section will select outgoing traffic
+- ability to set different confs on different proxy for the same route (overriding default conf for route without producer/consumer policies, on universal)
+- no need for specifying `spec.targetRef` when targeting `Mesh`, simplifies policy
+- gives more flexibility as you can target multiple outbounds in single policy. Reducing number of consumer policies
+- more consistent with current approach
+- easier policy matching code
+
+#### Disadvantages
+
+- can be harder to understand because `spec.targetRef` can be MeshSubset (which is optional and probably will rarely be used)
+- yet another migration for users (only for system policies as for namespaced policies this is already not allowed) 
+- mixing outbounds and routes in `spec.to[].targetRef`
+- hides the most often most important information under `to[]` (only in certain cases as most of the time policy will target Mesh and will have only `spec.to[].targetRef`)  
+
+### Leave Mesh*Routes in top level targetRef as it is now and introduce `spec.workloadTargetRef`
+
+#### Advantages
+
+- can be simpler with clear distinction of selecting proxies in `spec.workloadTargetRef` and other resources in `spec.targetRef`
+
+#### Disadvantages
+
+- mixing proxies, outbounds and routes in `spec.targetRef`
+- when `Mesh*Route` is in `spec.targetRef` targetRef we still have `spec.to[].targetRef` section which can be misleading as route already points to some outbound
```

Comment:
In fact this would make `spec.to` obsolete. Otherwise we can just make it optional like the other option suggests doing for `spec.targetRef`, so it's not really a disadvantage.

Reply:
right, removed this point as `spec.to[]` can be omitted in that case